distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -81,57 +73,97 @@ public static Action chunkedContent(String responseMessage, Duration messageDela
      * @return composite action that holds the headers required for a plain text response as well
      * as the response function itself.
      */
-    public static Action chunkedContentWithHeaders(String responseMessage, Duration messageDelay,
-            int numberOfFailures) {
-        return composite(getHeaders(),
-                chunkedContent(responseMessage, messageDelay, numberOfFailures));
+    private static Action createChunkedContent(String responseMessage,
+            Duration messageDelay, int numberOfFailures, HeaderCapture headerCapture) {
+
+        Action response = composite(getChunkedResponseHeaders(),
+                custom(new ChunkedContentFunction(responseMessage,
+                messageDelay,
+                numberOfFailures,
+                headerCapture)));
+
+        // adds the Accept-Ranges header for range-header support
+        if (headerCapture != null) {
```

Comment:
It is an option on the builder, `allowPartialContent()`. It takes a `HeaderCapture` in as a parameter and sets `headerCapture` equal to it, otherwise it stays equal to null. The null check is just a way to see if that option was enabled in the builder. I felt like adding a separate boolean variable in addition to setting the headerCapture object was unnecessary.